core/domain/state_domain.py

```diff
@@ -714,6 +714,190 @@ def validate(self):
+
+        Args:
+            content_translations_dict: dict. The dict representation of
+                ContentTranslations object.
+
+        Returns:
+            ContentTranslations. The corresponding ContentTranslations domain
+            object.
+        """
+        content_translations = {}
+        for (content_id, language_code_to_translation_script) in (
+                content_translations_dict.iteritems()):
+            content_translations[content_id] = {}
+            for (language_code, translation_script) in (
+                    language_code_to_translation_script.iteritems()):
+                content_translations[content_id][language_code] = (
+                    TranslationScript.from_dict(translation_script))
+
+        return cls(content_translations)
+
+    def validate(self):
+        """Validates properties of the ContentTranslations.
+
+        Raises:
+            ValidationError: One or more attributes of the ContentTranslations
+            are invalid.
+        """
+        for (content_id, language_code_to_translation_script) in (
+                self.content_translations.iteritems()):
+            if not isinstance(content_id, basestring):
+                raise utils.ValidationError(
+                    'Expected content_id to be a string, received %s'
+                    % content_id)
+            if not isinstance(language_code_to_translation_script, dict):
+                raise utils.ValidationError(
+                    'Expected content_id value to be a dict, received %s'
+                    % language_code_to_translation_script)
+            for (language_code, translation_script) in (
+                    language_code_to_translation_script.iteritems()):
+                if not isinstance(language_code, basestring):
+                    raise utils.ValidationError(
+                        'Expected language_code to be a string, received %s'
+                        % language_code)
+                if not utils.is_valid_language_code(language_code):
+                    raise utils.ValidationError(
+                        'Invalid language_code: %s' % language_code)
+                # TODO(DubeySandeep): Remove this check once App Engine supports
+                # 3-letter language codes in search.
+                if len(language_code) != 2:
+                    raise utils.ValidationError(
+                        'Invalid language_code, it should have exactly 2 '
+                        'letters: %s' % language_code)
+                translation_script.validate()
+
+    def get_available_languages(self):
+        """Returns a set of language available in the ContentTranslations.
+
+        Returns:
+            set. A set of languages available in the ContentTranslations.
+        """
+        languages_list = []
+        for (_, language_code_to_translation_script) in (
+                self.content_translations.iteritems()):
+            for (language_code, _) in (
```

Comment:
for language_code in language_code_to_translation_script

Reply:
Done!